auth/auth/auth.py

```diff
@@ -341,11 +318,13 @@ async def logout(request, userdata):
 @routes.get('/api/v1alpha/login')
 async def rest_login(request):
     callback_port = request.query['callback_port']
-
-    flow = get_flow(f'http://127.0.0.1:{callback_port}/oauth2callback')
-    authorization_url, state = flow.authorization_url(access_type='offline', include_granted_scopes='true')
-
-    return web.json_response({'authorization_url': authorization_url, 'state': state})
+    if CLOUD == 'azure':
+        host = 'localhost'
+    else:
+        assert CLOUD == 'gcp'
+        host = '127.0.0.1'
```

Comment:
This shouldn't differ between clouds? `127.0.0.1` is defined as the loopback address ([RFC1122 3.2.1.3 (g)](https://datatracker.ietf.org/doc/html/rfc1122#page-31) and [RFC6890 2.2.2](https://datatracker.ietf.org/doc/html/rfc6890#page-7)). Did 127.0.0.1 not work when using the Hail CLI to log into Azure?

Reply:
Azure won't let you use `127.0.0.1`. It breaks their formatting requirements for a URL. I can try again.

Reply:
This is the error you get. The URL has to match exactly

<img width="618" alt="Screen Shot 2021-12-15 at 12 12 12 PM" src="https://user-images.githubusercontent.com/1693348/146232967-0a06a4fa-c645-4067-a79c-53408ca096c2.png">

Reply:
> AADSTS50011: The reply URL specified in the request does not match the reply URLs configured for the application:

Reply:
Huh interesting, the portal requires that the reply URL use https unless the domain is specfically `localhost`. I tried to add 127.0.0.1 under https, does that work? Seems we should be able to use either the IP or domain for both clouds.

Reply:
ah right we run a local server for hailctl auth login

Reply:
If I change the Google one to 'localhost', I get a similar error message. The option is to keep this code or I go and duplicate all the callbacks in GCP to use localhost instead.

Reply:
Man, this chaps my ass. Let's add a comment noting that this is specifically because azure does not support `http://127.0.0.1`. Otherwise, I'm definitely going to forget and try to change this in a year.